core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java

```diff
@@ -364,6 +320,35 @@ private void createNewLogFile() throws IOException {
     LOG.info("Created current log file: {}", currentLog);
   }
 
+  private void completeLog(UfsJournalFile currentLog, long nextSequenceNumber) throws IOException {
+    String current = currentLog.getLocation().toString();
+    if (nextSequenceNumber <= currentLog.getStart()) {
+      LOG.info("No journal entry found in current journal file {}. Deleting it", current);
+      if (!mUfs.deleteFile(current)) {
+        LOG.warn("Failed to delete empty journal file {}", current);
+      }
+      return;
+    }
+    LOG.info("Completing log {} with next sequence number {}", current, nextSequenceNumber);
+    String completed = UfsJournalFile
+        .encodeLogFileLocation(mJournal, currentLog.getStart(), nextSequenceNumber).toString();
+
+    if (!mUfs.renameFile(current, completed)) {
+      // Renames could happen concurrently.
```

Comment:
could you add when these completes could happen concurrently?

Reply:
added to the javadoc